### PR TITLE
Improve parent lookup logging

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

Just some log improvements that would help in debugging something like https://github.com/sigp/lighthouse/pull/5432

### CODE REVIEW

File: beacon_node/http_api/src/publish_blocks.rs [hunk 1]
```diff
@@ -133,7 +133,7 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlockConten
                     log,
                     "Not publishing block - not gossip verified";
                     "slot" => slot,
-                    "error" => ?e
+                    "error" => %e
```
Comment (realbigsean) [inline beacon_node/http_api/src/publish_blocks.rs:136]: debug causes us to dump the entire block/blob on a `ParentUnknown` error, write just writes the parent root

**Review by jimmygchen — APPROVED** [review]
Looks good!

File: beacon_node/beacon_chain/tests/block_verification.rs [hunk 2]
```diff
@@ -1087,7 +1087,7 @@ async fn block_gossip_verification() {
     assert!(
         matches!(
             unwrap_err(harness.chain.verify_block_for_gossip(Arc::new(block.clone())).await),
-            BlockError::BlockIsAlreadyKnown,
+            BlockError::BlockIsAlreadyKnown(_),
```
Comment (jimmygchen) [inline beacon_node/beacon_chain/tests/block_verification.rs:1090]: Might be worth matching for block root since we have it?

Comment (realbigsean) [reply beacon_node/beacon_chain/tests/block_verification.rs:1090]: you can't use `matches!` with a value in the pattern you're matching, so I tried to swap this for `assert_eq!` but it requires deriving `PartialEq` on a bunch of types (which is why I think this assert is written as it is)

Comment (jimmygchen) [reply beacon_node/beacon_chain/tests/block_verification.rs:1090]: All good, that was a nitpick - was almost tempted to delete it lol

**Comment by realbigsean** [comment]
@mergifyio queue

**Pull request merged**